From e26f986d6ccfa5f42377562e9c5317d9c48c7dd1 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 17 Sep 2019 14:49:02 +0000
Subject: [PATCH 0892/1000] MANTIS 0055465 CUSS 1.5.1 Biometrics Interface

ignore *idl *.xsd in eclipse project

codes.idl V1.5.0
- added data status DS_TYPES_BIOMETRIC

added file CUSS-1.5.BIOMETRICS.xsd


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1484 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/.classpath       |   4 +-
 .../Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.xsd   | 395 ++++++++++++++++++
 SelfServiceCommon/Massai/idl/Cuss/codes.idl   |  10 +-
 3 files changed, 405 insertions(+), 4 deletions(-)
 create mode 100644 SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.xsd

diff --git a/SelfServiceCommon/Massai/idl/.classpath b/SelfServiceCommon/Massai/idl/.classpath
index 58020eef..fb33f9b0 100644
--- a/SelfServiceCommon/Massai/idl/.classpath
+++ b/SelfServiceCommon/Massai/idl/.classpath
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" path="Cuss"/>
-	<classpathentry kind="src" path="Massai"/>
+	<classpathentry including="*.idl|*.xsd" kind="src" path="Cuss"/>
+	<classpathentry including="*.idl|*.xsd" kind="src" path="Massai"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
 	<classpathentry kind="output" path="bin"/>
 </classpath>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.xsd b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.xsd
new file mode 100644
index 00000000..4ef556b8
--- /dev/null
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.xsd
@@ -0,0 +1,395 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Mit XMLSpy v2013 rel. 2 (http://www.altova.com) von Andreas Gehling (Materna GmbH) bearbeitet -->
+<xs:schema targetNamespace="http://www.iata.org/CUSS/biometrics-1.0" xmlns:bio="http://www.iata.org/CUSS/biometrics-1.0" xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified">
+	<xs:simpleType name="biometric-provider-msg-type">
+		<xs:annotation>
+			<xs:documentation>Biometric provider message (document) type.</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="none"/>
+			<xs:enumeration value="request"/>
+			<xs:enumeration value="request-data"/>
+			<xs:enumeration value="response"/>
+		</xs:restriction>
+	</xs:simpleType>
+	<xs:simpleType name="characteristics-document-type">
+		<xs:annotation>
+			<xs:documentation>Characteristics document type.</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="none"/>
+			<xs:enumeration value="request"/>
+			<xs:enumeration value="response"/>
+		</xs:restriction>
+	</xs:simpleType>
+	<xs:simpleType name="cu-msg-schema-version-type">
+		<xs:annotation>
+			<xs:documentation>Common use message schema version.
+Version format: 2-digit-major.2-digit-minor.4-digit-increment. New major number extends or changes schema and may not guarantee backward compatibility (may break existing implementations).
+New minor number extends schema and guarantees backward compatibility (does not break existing implementations).
+New increment number implements no functional changes (e.g. documentation changes, additional enumerated values, etc.).
+</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:minLength value="3"/>
+			<xs:maxLength value="10"/>
+			<xs:pattern value="[0-9]{1,2}\.[0-9]{1,2}(\.[0-9]{1,4}){0,1}"/>
+		</xs:restriction>
+	</xs:simpleType>
+	<xs:simpleType name="ds-types-type">
+		<xs:annotation>
+			<xs:documentation>DS type(s) string.</xs:documentation>
+		</xs:annotation>
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="DS_TYPES_BIOMETRIC_PROVIDER"/>
+		</xs:restriction>
+	</xs:simpleType>
+	<xs:element name="biometric-provider-msg">
+		<xs:annotation>
+			<xs:documentation>Common use biometric provider component request/response message.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:choice>
+				<xs:element name="biometric-request">
+					<xs:annotation>
+						<xs:documentation>Biometric function request from common use application to biometric provider component.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:sequence>
+							<xs:element name="environment" minOccurs="0">
+								<xs:annotation>
+									<xs:documentation>Execution environment and parameters.</xs:documentation>
+								</xs:annotation>
+								<xs:complexType>
+									<xs:sequence>
+										<xs:element name="accessibility-mode" type="xs:boolean" minOccurs="0">
+											<xs:annotation>
+												<xs:documentation>Whether to execute the biometric function in accessibility mode.</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+										<xs:element name="language">
+											<xs:annotation>
+												<xs:documentation>ISO 639-1 (2-character) code for the interface language to be used by the biometric provider.</xs:documentation>
+											</xs:annotation>
+											<xs:simpleType>
+												<xs:restriction base="xs:string">
+													<xs:length value="2"/>
+												</xs:restriction>
+											</xs:simpleType>
+										</xs:element>
+									</xs:sequence>
+								</xs:complexType>
+							</xs:element>
+							<xs:element name="agreements-list" minOccurs="0">
+								<xs:annotation>
+									<xs:documentation>List of agreements/consents to present prior to or during enrollment.</xs:documentation>
+								</xs:annotation>
+								<xs:complexType>
+									<xs:sequence>
+										<xs:element name="agreement-text" maxOccurs="unbounded">
+											<xs:annotation>
+												<xs:documentation>Individual agreement/consent text to present to the potential enrollee.</xs:documentation>
+											</xs:annotation>
+											<xs:complexType>
+												<xs:simpleContent>
+													<xs:extension base="xs:string">
+														<xs:attribute name="agreement-name" type="xs:string" use="optional">
+															<xs:annotation>
+																<xs:documentation>Name or title of consent or agreement.</xs:documentation>
+															</xs:annotation>
+														</xs:attribute>
+														<xs:attribute name="acceptance-mandatory" type="xs:boolean" use="required">
+															<xs:annotation>
+																<xs:documentation>Whether agreement or consent is mandatory for the enrollment or other biometric function to proceed.</xs:documentation>
+															</xs:annotation>
+														</xs:attribute>
+													</xs:extension>
+												</xs:simpleContent>
+											</xs:complexType>
+										</xs:element>
+									</xs:sequence>
+								</xs:complexType>
+							</xs:element>
+							<xs:element ref="bio:passenger" minOccurs="0"/>
+							<xs:element ref="bio:gp-data-list" minOccurs="0"/>
+						</xs:sequence>
+					</xs:complexType>
+				</xs:element>
+				<xs:element name="biometric-request-data">
+					<xs:annotation>
+						<xs:documentation>Supplementary device-acquired or other unsolicited data received and forwarded by application during biometric provider function execution.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:choice>
+							<xs:element name="processing">
+								<xs:complexType>
+									<xs:choice>
+										<xs:element name="keypad-key" type="xs:string" maxOccurs="unbounded">
+											<xs:annotation>
+												<xs:documentation>A sequence of one or more keypad key identifiers as defined in the CUSS specification "Keyboard and Navigation Keypad" section. Values include, but are not limited to, NAVUP, NAVDOWN, NAVPREVIOUS, NAVNEXT, NAVENTER, NAVHOME, NAVEND and NAVHELP.</xs:documentation>
+											</xs:annotation>
+										</xs:element>
+									</xs:choice>
+								</xs:complexType>
+							</xs:element>
+							<xs:element ref="bio:passenger"/>
+						</xs:choice>
+					</xs:complexType>
+				</xs:element>
+				<xs:element name="biometric-response">
+					<xs:annotation>
+						<xs:documentation>Function request response from biometric provider component to common use application.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:sequence>
+							<xs:element ref="bio:passenger" minOccurs="0"/>
+							<xs:element ref="bio:gp-data-list" minOccurs="0"/>
+						</xs:sequence>
+						<xs:attribute name="response-type" use="required">
+							<xs:simpleType>
+								<xs:restriction base="xs:string">
+									<xs:enumeration value="cancellation"/>
+									<xs:enumeration value="failure"/>
+									<xs:enumeration value="success"/>
+								</xs:restriction>
+							</xs:simpleType>
+						</xs:attribute>
+					</xs:complexType>
+				</xs:element>
+			</xs:choice>
+			<xs:attribute name="biometric-provider-msg-type" type="bio:biometric-provider-msg-type" use="required">
+				<xs:annotation>
+					<xs:documentation>What type of biometric provider message this is.</xs:documentation>
+				</xs:annotation>
+			</xs:attribute>
+			<xs:attribute name="biometric-provider-msg-return-type" type="bio:biometric-provider-msg-type" use="required">
+				<xs:annotation>
+					<xs:documentation>What type of biometric provider message is expected to be returned.</xs:documentation>
+				</xs:annotation>
+			</xs:attribute>
+			<xs:attribute name="biometric-function-type" use="required">
+				<xs:simpleType>
+					<xs:restriction base="xs:string">
+						<xs:enumeration value="enrol"/>
+						<xs:enumeration value="identify"/>
+						<xs:enumeration value="verify"/>
+					</xs:restriction>
+				</xs:simpleType>
+			</xs:attribute>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="cu-msg">
+		<xs:annotation>
+			<xs:documentation>Common use message document root element.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:choice>
+				<xs:element name="characteristics">
+					<xs:annotation>
+						<xs:documentation>Component characteristics.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:choice>
+							<xs:element name="characteristics-request" minOccurs="0">
+								<xs:annotation>
+									<xs:documentation>Unqualified request for characteristics response to be returned. The presence of this element is optional as it contains no further information in addition to that indicated in the parent elemtent's characteristics-document-type attribute.</xs:documentation>
+								</xs:annotation>
+							</xs:element>
+							<xs:element name="characteristics-response">
+								<xs:complexType>
+									<xs:sequence>
+										<xs:element name="cu-msg-schema-version-list">
+											<xs:annotation>
+												<xs:documentation>Supported common use message schema version list. Each supported version must be explicitly listed.</xs:documentation>
+											</xs:annotation>
+											<xs:complexType>
+												<xs:sequence>
+													<xs:element name="cu-msg-schema-version" type="bio:cu-msg-schema-version-type" maxOccurs="unbounded">
+														<xs:annotation>
+															<xs:documentation>Supported common use message schema version.</xs:documentation>
+														</xs:annotation>
+													</xs:element>
+												</xs:sequence>
+											</xs:complexType>
+										</xs:element>
+									</xs:sequence>
+									<xs:attribute name="ds-types" type="bio:ds-types-type" use="required">
+										<xs:annotation>
+											<xs:documentation>CUSS-defined component DS_TYPES indicator.</xs:documentation>
+										</xs:annotation>
+									</xs:attribute>
+								</xs:complexType>
+							</xs:element>
+						</xs:choice>
+						<xs:attribute name="characteristics-document-type" type="bio:characteristics-document-type" use="required">
+							<xs:annotation>
+								<xs:documentation>What type of biometric provider characteristics document this is.</xs:documentation>
+							</xs:annotation>
+						</xs:attribute>
+						<xs:attribute name="characteristics-document-return-type" type="bio:characteristics-document-type" use="required">
+							<xs:annotation>
+								<xs:documentation>What type of biometric provider characteristics document is expected to be returned.</xs:documentation>
+							</xs:annotation>
+						</xs:attribute>
+					</xs:complexType>
+				</xs:element>
+				<xs:element ref="bio:biometric-provider-msg"/>
+				<xs:element name="error-response">
+					<xs:annotation>
+						<xs:documentation>Document cannot be parsed or validated, or is unexpected, out-of-sequence or out-of-context for the current component state.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:attribute name="error-document" type="xs:string" use="required">
+							<xs:annotation>
+								<xs:documentation>Echo of unreadable or illogical document in base64 format.</xs:documentation>
+							</xs:annotation>
+						</xs:attribute>
+						<xs:attribute name="error-reason" use="required">
+							<xs:annotation>
+								<xs:documentation>Human-readable reason that the document could not be processed by the component.</xs:documentation>
+							</xs:annotation>
+							<xs:simpleType>
+								<xs:restriction base="xs:string">
+									<xs:enumeration value="data-error"/>
+									<xs:enumeration value="format-error"/>
+									<xs:enumeration value="illogical"/>
+								</xs:restriction>
+							</xs:simpleType>
+						</xs:attribute>
+					</xs:complexType>
+				</xs:element>
+			</xs:choice>
+			<xs:attribute name="cu-msg-schema-version" use="required">
+				<xs:annotation>
+					<xs:documentation>Version format: 2-digit-major.2-digit-minor.4-digit-increment.
+New major number extends or changes schema and may not guarantee backward compatibility (may break existing implementations).
+New minor number extends schema and guarantees backward compatibility (does not break existing implementations).
+New increment number implements no functional changes (e.g. documentation changes, additional enumerated values, etc.).
+</xs:documentation>
+				</xs:annotation>
+				<xs:simpleType>
+					<xs:restriction base="bio:cu-msg-schema-version-type">
+						<xs:minLength value="3"/>
+						<xs:maxLength value="10"/>
+					</xs:restriction>
+				</xs:simpleType>
+			</xs:attribute>
+			<xs:attribute name="cu-msg-session-id" use="optional">
+				<xs:annotation>
+					<xs:documentation>An arbitrary tracking value that the CUSS application chooses and assigns when starting a biometric provider function. The CUSS platform must then echo this requested value in all subsequent responses or asynchronous event messages related to that biometric provider function. This correlation permits a CUSS application to match and associate request and response messages and is critical for the case where an application has more than one multi-step biometric provider function ongoing concurrently.</xs:documentation>
+				</xs:annotation>
+				<xs:simpleType>
+					<xs:restriction base="xs:string"/>
+				</xs:simpleType>
+			</xs:attribute>
+			<xs:attribute name="cu-msg-document-type" use="required">
+				<xs:simpleType>
+					<xs:restriction base="xs:string">
+						<xs:enumeration value="characteristics"/>
+						<xs:enumeration value="biometric-provider-msg"/>
+						<xs:enumeration value="error-response"/>
+					</xs:restriction>
+				</xs:simpleType>
+			</xs:attribute>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="gp-data-item">
+		<xs:annotation>
+			<xs:documentation>General purpose biometric function request and/or response individual data item.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:simpleContent>
+				<xs:extension base="xs:string">
+					<xs:attribute name="data-item-name" type="xs:string" use="required"/>
+				</xs:extension>
+			</xs:simpleContent>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="gp-data-list">
+		<xs:annotation>
+			<xs:documentation>General purpose biometric function request and/or response data item list.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:sequence>
+				<xs:element ref="bio:gp-data-item" maxOccurs="unbounded"/>
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+	<xs:element name="passenger">
+		<xs:annotation>
+			<xs:documentation>Biometric function request passenger data.</xs:documentation>
+		</xs:annotation>
+		<xs:complexType>
+			<xs:sequence>
+				<xs:element name="passenger-id" type="xs:string" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Airline's unique passenger identifiier.</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="passenger-family-name" type="xs:string" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Passenger's family name (last name or surname).</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="passenger-given-names" type="xs:string" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Passenger's given name (christian or first names).</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="passenger-mrz-data" type="xs:string" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>MRZ data from passenger's government-issued travel, visa or residency document.</xs:documentation>
+					</xs:annotation>
+				</xs:element>
+				<xs:element name="passenger-biometrics" minOccurs="0">
+					<xs:annotation>
+						<xs:documentation>Passenger's biometrics image data.</xs:documentation>
+					</xs:annotation>
+					<xs:complexType>
+						<xs:simpleContent>
+							<xs:extension base="xs:hexBinary">
+								<xs:attribute name="biometrics-type" use="required">
+									<xs:annotation>
+										<xs:documentation>Biometrics type.</xs:documentation>
+									</xs:annotation>
+									<xs:simpleType>
+										<xs:restriction base="xs:string">
+											<xs:enumeration value="face"/>
+											<xs:enumeration value="finger"/>
+											<xs:enumeration value="iris"/>
+											<xs:enumeration value="palm"/>
+										</xs:restriction>
+									</xs:simpleType>
+								</xs:attribute>
+								<xs:attribute name="light-type" use="optional">
+									<xs:annotation>
+										<xs:documentation>Image light type.</xs:documentation>
+									</xs:annotation>
+									<xs:simpleType>
+										<xs:restriction base="xs:string">
+											<xs:enumeration value="ir"/>
+											<xs:enumeration value="uv"/>
+											<xs:enumeration value="visible"/>
+										</xs:restriction>
+									</xs:simpleType>
+								</xs:attribute>
+								<xs:attribute name="data-format" use="required">
+									<xs:annotation>
+										<xs:documentation>Image data format.</xs:documentation>
+									</xs:annotation>
+									<xs:simpleType>
+										<xs:restriction base="xs:string">
+											<xs:enumeration value="bmp"/>
+											<xs:enumeration value="jpg"/>
+											<xs:enumeration value="png"/>
+										</xs:restriction>
+									</xs:simpleType>
+								</xs:attribute>
+							</xs:extension>
+						</xs:simpleContent>
+					</xs:complexType>
+				</xs:element>
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+</xs:schema>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/codes.idl b/SelfServiceCommon/Massai/idl/Cuss/codes.idl
index e8947cab..8810abfe 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/codes.idl
+++ b/SelfServiceCommon/Massai/idl/Cuss/codes.idl
@@ -6,7 +6,7 @@
 //
 //  Date:      31.03.2015
 //
-//  Version:   1.4.5
+//  Version:   1.5.0
 //
 //  Author:    IATA Passenger Experience Management Group: CUWG CUSS-TSG
 //
@@ -93,6 +93,11 @@ Changed:   Added baggage handling status codes (Nicolas Bruns, 29.06.2004)
 @li Mantis 0054609: CUSS 1.5 codes.idl: DS_TYPES_SSML11, DS_TYPES_KEY_UP, DS_TYPES_KEY_DOWN
 - fixed DS_TYPES_KEY_UP, DS_TYPES_KEY_DOWN
 
+@date 04.09.2019
+@li V1.5.0
+@li Mantis 0055465: CUSS 1.5.1 Biometrics Interface
+- added data status DS_TYPES_BIOMETRIC
+
 
 */
 
@@ -222,7 +227,8 @@ module datastatus
     const long DS_TYPES_KEY                      = 26000; /**< key pressed */
     const long DS_TYPES_KEY_UP                   = 26010; /**< Navigation keypad event supported by Keypad Interface */
     const long DS_TYPES_KEY_DOWN                 = 26020; /**< Navigation keypad event supported by Keypad Interface */
-    
+
+    const long DS_TYPES_BIOMETRIC                = 27000; /**< Biometric Interface: BiomerticProvider.xsd */
 
     const long DS_TYPES_CYPHER_APP_CERTIFICATE   = 40010; /**< certificate of the application */
     const long DS_TYPES_CYPHER_HW_CERTIFICATE    = 40020; /**< certificate of the hardware */
-- 
2.41.0.windows.1

